pkg/storegateway/series_refs_test.go

```diff
@@ -1387,6 +1450,110 @@ func TestOpenBlockSeriesChunkRefsSetsIterator(t *testing.T) {
 	}
 }
 
+func TestOpenBlockSeriesChunkRefsSetsIterator_SeriesCaching(t *testing.T) {
+	existingSeries := []labels.Labels{
+		labels.FromStrings("a", "1", "b", "1"),
+		labels.FromStrings("a", "1", "b", "2"),
+		labels.FromStrings("a", "2", "b", "1"),
+		labels.FromStrings("a", "2", "b", "2"),
+		labels.FromStrings("a", "3", "b", "1"),
+		labels.FromStrings("a", "3", "b", "2"),
+	}
+	newTestBlock := prepareTestBlock(test.NewTB(t), func(tb testing.TB, appender storage.Appender) {
+		for ts := int64(0); ts < 10; ts++ {
+			for _, s := range existingSeries {
+				_, err := appender.Append(0, s, ts, 0)
+				assert.NoError(t, err)
+			}
+		}
+		assert.NoError(t, appender.Commit())
+	})
+
+	matchers := []*labels.Matcher{
+		labels.MustNewMatcher(labels.MatchRegexp, "a", ".+"),
+	}
+
+	for batchSize := 1; batchSize < len(existingSeries)*2; batchSize++ {
```

Comment:
[nit] Isn't +1 enough to test the case the batchSize is bigger than the actual series? 🤔 

```suggestion
	for batchSize := 1; batchSize < len(existingSeries) + 1; batchSize++ {
```

Reply:
👍 changed in 816ddd3705390bc4ead1c5d1c2354945986943aa